html/pfappserver/root/static.alt/src/views/Configuration/_components/AuthenticationSourcesView.vue

```diff
@@ -0,0 +1,153 @@
+import pfMixinCtrlKey from '@/components/pfMixinCtrlKey'
+import pfMixinEscapeKey from '@/components/pfMixinEscapeKey'
+import {
+  pfConfigurationAuthenticationSourcesViewFields as fields,
+  pfConfigurationAuthenticationSourcesViewDefaults as defaults
+} from '@/globals/pfConfiguration'
+const { validationMixin } = require('vuelidate')
+
+export default {
+  name: 'AuthenticationSourcesView',
+  mixins: [
+    validationMixin,
+    pfMixinCtrlKey,
+    pfMixinEscapeKey
+  ],
+  components: {
+    pfConfigView,
+    pfButtonSave,
+    pfButtonDelete
+  },
+  props: {
+    storeName: { // from router
+      type: String,
+      default: null,
+      required: true
+    },
+    sourceType: { // from router (or source)
+      type: String,
+      default: null
+    },
+    isNew: { // from router
+      type: Boolean,
+      default: false
+    },
+    isClone: { // from router
+      type: Boolean,
+      default: false
+    },
+    id: { // from router
+      type: String,
+      default: null
+    }
+  },
+  data () {
+    return {
+      realms: [], // all realms
+      sources: [], // all sources
+      source: defaults(this), // will be overloaded with the data from the store
+      sourceValidations: {} // will be overloaded with data from the pfConfigView
+    }
+  },
+  validations () {
+    return {
+      source: this.sourceValidations
+    }
+  },
+  computed: {
+    isLoading () {
+      return this.$store.getters['$_sources/isLoading']
+    },
+    invalidForm () {
+      return this.$v.source.$invalid || this.$store.getters['$_sources/isWaiting']
+    },
+    getForm () {
+      return {
+        labelCols: 3,
+        fields: fields(this)
+      }
+    }
+  },
+  methods: {
+    close (event) {
+      this.$router.push({ name: 'sources' })
+    },
+    create (event) {
+      const ctrlKey = this.ctrlKey
+      this.$store.dispatch('$_sources/createAuthenticationSource', this.source).then(response => {
+        if (ctrlKey) { // [CTRL] key pressed
+          this.close()
+        } else {
+          this.$router.push({ name: 'source', params: { id: this.source.id } })
+        }
+      })
+    },
+    save (event) {
+      const ctrlKey = this.ctrlKey
+      this.$store.dispatch('$_sources/updateAuthenticationSource', this.source).then(response => {
+        if (ctrlKey) { // [CTRL] key pressed
+          this.close()
+        }
+      })
+    },
+    remove (event) {
+      this.$store.dispatch('$_sources/deleteAuthenticationSource', this.id).then(response => {
+        this.close()
+      })
+    }
+  },
+  created () {
+    if (this.id) {
+      this.$store.dispatch('$_sources/getAuthenticationSource', this.id).then(data => {
+        this.sourceType = data.type
+        this.source = Object.assign({}, data)
+        if (this.isClone) {
+          this.source.id = null
+        }
+      })
+    }
+    this.source.type = this.sourceType
+    this.$store.dispatch('$_sources/all').then(data => {
+      this.sources = data
```

Comment:
Where is this used?

Reply:
It's used when setting up a SAML source: https://github.com/inverse-inc/packetfence/pull/3774/files#diff-36c680975adb5064cbf6ccd8cf8ed28fR438